pkg/controller/validate/validate.go

```diff
@@ -86,14 +77,7 @@ func deploymentForGov(deployment *mdbv1.AtlasDeploymentSpec, regionUsageRestrict
 	var err error
 
 	if deployment.DeploymentSpec != nil {
-		regionErr := validCloudGovRegion(regionUsageRestrictions, deployment.DeploymentSpec.ProviderSettings.RegionName)
```

Comment:
not sure why this gov related check is no longer needed here, can you elaborate?

Reply:
Since there is no more legacy clusters, regions have to validated for every replica set. See this methods is called below.